rs/nns/governance/src/voting.rs

```diff
@@ -190,47 +460,79 @@ impl ProposalVotingStateMachine {
         &mut self,
         neuron_store: &mut NeuronStore,
         ballots: &mut HashMap<u64, Ballot>,
+        is_over_instructions_limit: fn() -> bool,
     ) {
-        while let Some(neuron_id) = self.neurons_to_check_followers.pop_first() {
-            self.add_followers_to_check(neuron_store, neuron_id, self.topic);
-        }
-
-        // Memory optimization, will not cause tests to fail if removed
-        retain_neurons_with_castable_ballots(&mut self.followers_to_check, ballots);
-
-        while let Some(follower) = self.followers_to_check.pop_first() {
-            let vote = match neuron_store.neuron_would_follow_ballots(follower, self.topic, ballots)
-            {
-                Ok(vote) => vote,
-                Err(e) => {
-                    // This is a bad inconsistency, but there is
-                    // nothing that can be done about it at this
-                    // place.  We somehow have followers recorded that don't exist.
-                    eprintln!("error in cast_vote_and_cascade_follow when gathering induction votes: {:?}", e);
-                    Vote::Unspecified
+        let mut actions_performed = 0;
+        let check_after_number_actions = 50;
```

Comment:
We should remove this, but it's going to make the benchmarks look bad due to the mechanics mentioned in above thread...